test/cat.jl

```diff
@@ -137,15 +116,121 @@ module TestCat
     @test isequal(vcat(dtd, dtc0, dtc), dc)
     @test eltypes(vcat(dtd, dtc0)) == eltypes(dc)
 
-    # Missing columns
-    rename!(dtd, :a, :b)
-    dtda = DataTable(b = NullableArray(Nullable{Int}[2, 3, 4, Nullable(), Nullable(), Nullable()]),
-                     a = NullableCategoricalVector(Nullable{Int}[Nullable(), Nullable(), Nullable(), 1, 2, 2]))
-    @test isequal(vcat(dtd, dta), dtda)
-
-    # Alignment
-    @test isequal(vcat(dtda, dtd, dta), vcat(dtda, dtda))
-
     # vcat should be able to concatenate different implementations of AbstractDataTable (PR #944)
     @test isequal(vcat(view(DataTable(A=1:3),2),DataTable(A=4:5)), DataTable(A=[2,4,5]))
+
+    @testset "vcat >2 args" begin
+        @test vcat(DataTable(), DataTable(), DataTable()) == DataTable()
+        dt = DataTable(x = trues(1), y = falses(1))
+        @test vcat(dt, dt, dt) == DataTable(x = trues(3), y = falses(3))
+    end
+
+    @testset "vcat mixed coltypes" begin
+        dt = vcat(DataTable([[1]], [:x]), DataTable([[1.0]], [:x]))
+        @test dt == DataTable([[1.0, 1.0]], [:x])
+        @test eltypes(dt)[1] == Float64
+        dt = vcat(DataTable([[1]], [:x]), DataTable([["1"]], [:x]))
+        @test dt == DataTable([[1, "1"]], [:x])
+        @test eltypes(dt)[1] == Any
+        dt = vcat(DataTable([NullableArray([1])], [:x]), DataTable([[1]], [:x]))
+        @test dt == DataTable([NullableArray([1, 1])], [:x])
+        @test eltypes(dt)[1] == Nullable{Int}
```

Comment:
Can you rather test the type of the column, to make sure we get a `NullableArray` rather than an `Array{Nullable}`? Same below for `(Nullable)CategoricalArray` vs. Array{CategoricalValue}`.